--- zip.c.orig	2013-02-24 08:32:30.000000000 +0100
+++ zip.c	2014-05-24 10:36:41.744297397 +0200
@@ -1879,6 +1879,18 @@
   return err;
 }
 
+extern int ZEXPORT zipFlush (zipFile file)
+{
+    zip64_internal* zi;
+
+    if (file == NULL)
+        return ZIP_PARAMERROR;
+
+    zi = (zip64_internal*)file;
+
+    return ZFLUSH64(zi->z_filefunc,zi->filestream);
+}
+
 extern int ZEXPORT zipClose (zipFile file, const char* global_comment)
 {
     zip64_internal* zi;
