--- ioapi.h.orig	2014-05-24 09:54:26.302884145 +0200
+++ ioapi.h	2014-05-24 09:47:36.375502868 +0200
@@ -135,6 +135,7 @@
 typedef voidpf   (ZCALLBACK *open_file_func)      OF((voidpf opaque, const char* filename, int mode));
 typedef uLong    (ZCALLBACK *read_file_func)      OF((voidpf opaque, voidpf stream, void* buf, uLong size));
 typedef uLong    (ZCALLBACK *write_file_func)     OF((voidpf opaque, voidpf stream, const void* buf, uLong size));
+typedef int      (ZCALLBACK *flush_file_func)     OF((voidpf opaque, voidpf stream));
 typedef int      (ZCALLBACK *close_file_func)     OF((voidpf opaque, voidpf stream));
 typedef int      (ZCALLBACK *testerror_file_func) OF((voidpf opaque, voidpf stream));
 
@@ -186,6 +187,7 @@
 
 #define ZREAD64(filefunc,filestream,buf,size)     ((*((filefunc).zfile_func64.zread_file))   ((filefunc).zfile_func64.opaque,filestream,buf,size))
 #define ZWRITE64(filefunc,filestream,buf,size)    ((*((filefunc).zfile_func64.zwrite_file))  ((filefunc).zfile_func64.opaque,filestream,buf,size))
+#define ZFLUSH(filefunc,filestream)               ((*((filefunc).zflush_file))               ((filefunc).opaque,filestream))
 //#define ZTELL64(filefunc,filestream)            ((*((filefunc).ztell64_file)) ((filefunc).opaque,filestream))
 //#define ZSEEK64(filefunc,filestream,pos,mode)   ((*((filefunc).zseek64_file)) ((filefunc).opaque,filestream,pos,mode))
 #define ZCLOSE64(filefunc,filestream)             ((*((filefunc).zfile_func64.zclose_file))  ((filefunc).zfile_func64.opaque,filestream))
