--- unzip.c.orig	2014-05-23 15:29:40.975965289 +0200
+++ unzip.c	2014-05-23 15:29:40.979965126 +0200
@@ -35,9 +35,15 @@
  */
 
 
+#ifndef _WIN32_WCE
 #include <stdio.h>
 #include <stdlib.h>
+#else
+#include <stdio.h>
+#include "celib.h"
+#endif
 #include <string.h>
+
 #include "zlib.h"
 #include "unzip.h"
 
@@ -1462,7 +1468,7 @@
 
     if (ZREAD(pfile_in_zip_read_info->z_filefunc,
               pfile_in_zip_read_info->filestream,
-              buf,size_to_read)!=size_to_read)
+              buf,read_now)!=read_now)
         return UNZ_ERRNO;
 
     return (int)read_now;
