--- zip.c.orig	2014-05-24 09:54:26.314883072 +0200
+++ zip.c	2014-05-24 09:52:59.338410979 +0200
@@ -1879,6 +1879,19 @@
   return err;
 }
 
+extern int ZEXPORT zipFlush (zipFile file)
+{
+    zip_internal* zi;
+    if (file == NULL)
+        return ZIP_PARAMERROR;
+    zi = (zip_internal*)file;
+
+    if (zi->z_filefunc.zflush_file != NULL && zi->filestream != NULL)
+      return ZFLUSH(zi->z_filefunc,zi->filestream);
+    else
+      return EOF;
+}
+
 extern int ZEXPORT zipClose (zipFile file, const char* global_comment)
 {
     zip64_internal* zi;
